### PR TITLE
Fix official build

### PR DESCRIPTION
**Pull request by genlu**
@tmat Would this work?

Also, yaml is HARD.

### CODE REVIEW

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -38,21 +38,26 @@ stages:
...
-    - powershell: Write-Host "##vso[task.setvariable variable=IsPRValidationBuild]($(PRNumber) -ne '') -and ($(PRNumber) -ne 'default')"
-      displayName: Setting IsPRValidationBuild variable
-
     - task: tagBuildOrRelease@0
-      displayName: Tag build
+      displayName: Tag official build
+      inputs:
+        type: 'Build'
+        tags: 'OfficialBuild'
+      condition: and(succeeded(), eq(variables['PRNumber'], 'default'))
```
Comment (JoeRobich) [inline azure-pipelines-official.yml]: Would be nice if PRNumber could be '' or 'default', but hopefully people will just leave the variable 'default' when queuing.

File: azure-pipelines-official.yml [hunk 2]
```diff
@@ -38,21 +38,26 @@ stages:
...
         type: 'Build'
-        tags: if (variables['IsPRValidationBuild']) { 'PRValidationBuild' } else { "OfficialBuild" }
+        tags: 'PRValidationBuild'
+      condition: and(succeeded(), ne(variables['PRNumber'], 'default'))
 
     - task: PowerShell@2
       displayName: Setup branch for insertion validation
       inputs:
         filePath: 'eng\setup-pr-validation.ps1'
         arguments: '-sourceBranchName $(SourceBranchName) -prNumber $(PRNumber)'
```
Comment (JoeRobich) [inline azure-pipelines-official.yml]: Has this been tested with an invalid PR number and does the pipeline fail at this point?

Comment (genlu) [reply azure-pipelines-official.yml]: This doesn't check for invalid number (with the assumption that the build should be tag as validation for any value other than 'default', including ''). The build will fail the next step

File: azure-pipelines-official.yml [hunk 3]
```diff
@@ -262,10 +267,25 @@ stages:
       queue:
         name: Hosted VS2017
 
-- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), eq(variables['IsPRValidationBuild'], false)) }}:
+# We need to skip post-build stages for PR validation build, but it can only be identified by 
+# the runtime variable 'PRNumber', thus this dummy stage. Also the dummy job is required
+# otherwise AzDO would just repeat jobs from previous stage.
```
Comment (genlu) [inline azure-pipelines-official.yml]: This isn't pretty, but it's the only way I can make it work...

**Comment by genlu** [comment]
I have validated this change, will merge once CI is done.

**Pull request merged**